core/templates/services/playthrough.service.ts

```diff
@@ -114,12 +117,16 @@ class CyclicStateTransitionsTracker {
 }
 
 class EarlyQuitTracker {
-  private stateName: string = null;
-  private expDurationInSecs: number = null;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  private stateName!: string;
+  private expDurationInSecs!: number;
 
   foundAnIssue(): boolean {
+    // TODO(#15212): Remove the below check once codebase is strictly typed.
```

Comment:
Be explicit about what the "below check" means, since there are 2 checks here.

Reply:
Done